src/artemis/stores/mapgraph/selections.cljs

```diff
@@ -100,11 +100,12 @@
   [entities join-expr lookup-ref selection]
   (if-not (type-cond? selection)
     join-expr
-    (reduce (fn [acc [condition selection]]
-              (if (= (:type-name (:type-condition condition))
-                     (:__typename (get entities (:artemis.mapgraph/ref lookup-ref))))
-                (reduced (into acc selection))
-                acc))
-            []
-            ;; converting map to tuples for easier access of individual key/val
-            (mapcat identity join-expr))))
+    (let [outer-selections (filter :field-name join-expr)] ; fields selected outside of the union fragment
+      (reduce (fn [acc [condition selection]]
+                (if (= (:type-name (:type-condition condition))
+                       (:__typename (get entities (:artemis.mapgraph/ref lookup-ref))))
+                  (reduced (into acc selection))
+                  acc))
+              outer-selections
```

Comment:
here we merge the selection set of the fragment with the fields that were selected outside of the fragment (that aren't fragments themselves)